From 70a3c0473c3d28acb195a09d2b864cd2746f20ee Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 19 Mar 2015 16:05:12 +0000
Subject: [PATCH 0649/1076] MANTIS 0035986 extension: appControl: some
 applications are visible when CLA is active

massai_systools.lib
V1.0.0.27

- added methods mSysWindow::setTopMost



git-svn-id: svn://localhost/SelfServiceCommon/trunk@831 90b65887-3827-0410-9a23-83215b262276
---
 .../Interfaces/inc/systools/mSysWindow.h        |  1 +
 .../Massai/cpp/Systools/src/local_def.cpp       | 17 +++++++++++------
 .../Massai/cpp/Systools/src/mSysWindowW32.cpp   |  2 +-
 3 files changed, 13 insertions(+), 7 deletions(-)

diff --git a/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h b/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h
index b99e4b8c..b46ae6c2 100644
--- a/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h
+++ b/SelfServiceCommon/Interfaces/inc/systools/mSysWindow.h
@@ -26,6 +26,7 @@ namespace mSysWindow
       virtual void maximize()= 0;
       virtual void minimize()= 0;
       virtual void showDefault()= 0;
+      virtual void setTopMost()= 0;
       virtual void show()= 0;
       virtual void hide()= 0;
       virtual void close()= 0;
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
index b6af7e05..4f16f8cd 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/local_def.cpp
@@ -1,6 +1,6 @@
 #include "tools/versionlib.h"
 
-VERSION_ID_DT(systools, 1, 0, 0, 26);
+VERSION_ID_DT(systools, 1, 0, 0, 27);
 
 /*!
 \file
@@ -8,19 +8,24 @@ VERSION_ID_DT(systools, 1, 0, 0, 26);
 
 \page massai_systools.lib Release Notes - massai_systools.lib
 
-@version 1.0.0.26
+@version 1.0.0.27
 
 \section changehistory Change history
 
 massai_systools.lib
 
+\version 1.0.0.27
+\li 19.03.2015
+\li extension SteT MANTIS 0035986: appControl: some applications are visible when CLA is active
+- added methods mSysWindow::setTopMost
+
 \version 1.0.0.26
 \li 27.02.2015
 \li extension SteT MANTIS 0035724: Desko-Penta: RFID; optimize antenna settings
--added methods mSysSmartCardReader::CardReader to access card reader direct,
- -virtual ResultCode connectReader() = 0;
- -virtual ResultCode disconnectReader() = 0;
- -virtual ResultCode control(unsigned int command, const std::string& data, std::string& response) = 0;
+- added methods mSysSmartCardReader::CardReader to access card reader direct,
+ - virtual ResultCode connectReader() = 0;
+ - virtual ResultCode disconnectReader() = 0;
+ - virtual ResultCode control(unsigned int command, const std::string& data, std::string& response) = 0;
 
 \version 1.0.0.25
 \li 26.11.2014
diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp
index 9019b6e9..2824ac32 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysWindowW32.cpp
@@ -36,6 +36,7 @@ namespace
       virtual std::string getClass();
       virtual std::string getProcessId();
       virtual bool isClosed();
+      virtual void setTopMost();
 
       DWORD getPID() const;
     private:
@@ -52,7 +53,6 @@ namespace
       );
       bool isMaximized();
       bool isTopMost();
-      void setTopMost();
       bool switchToWindow();
       bool setForeGroundWindow();
       void simulateInput();
-- 
2.41.0.windows.1

